src/selectors/entities/groups.ts

```diff
@@ -168,11 +168,16 @@ export const getAllAssociatedGroupsForReference: (state: GlobalState) => Group[]
     },
 );
 
-export const getCurrentUserGroupMentionKeys: (state: GlobalState) => UserMentionKey[] = createSelector(
+export const getCurrentUserGroupMentionKeys: (state: GlobalState, groupMentionHighlightDisabled: boolean) => UserMentionKey[] = createSelector(
     getAllAssociatedGroupsForReference,
-    (groups: Array<Group>) => {
+    (state: GlobalState, groupMentionHighlightDisabled = false) => groupMentionHighlightDisabled,
+    (groups: Array<Group>, groupMentionHighlightDisabled) => {
         const keys: UserMentionKey[] = [];
-        groups.forEach((group) => keys.push({key: `@${group.name}`}));
+
+        if (!groupMentionHighlightDisabled) {
+            groups.forEach((group) => keys.push({key: `@${group.name}`}));
+        }
+
```

Comment:
I dont think you want to be doing this - this will disable highlight only for the user without the group mentions permission not for other people.

Reply:
I'm confused then. Maybe I didn't understand the requirement correctly? So as a guest user if I don't have permission to mention a group and I mention a group, shouldn't it not be highlighted for only me? Do you have suggestions what the right way to tackle this problem is?

Reply:
It seems that it will be better for me to address this issue in another PR since we have the issue on Jira. So I reverted the changes I made and I'll be creating a new PR and we can discuss the solution there.